src/sonic-yang-models/yang-models/sonic-pac.yang

```diff
@@ -0,0 +1,237 @@
+                           a PAC controlled port.";
+                }
+
+                leaf reauth_enable {
+                    type boolean;
+                    description
+                        "Indicates whether Reauthentication is enabled on
+                            the port.";
+                }
+
+                leaf reauth_period {
+                    type uint32 {
+                        range 1..65535 {
+                            error-message "reauth period value must be in range of 1-65535.";
+                            error-app-tag reauth-period-invalid;
+                        }
+                    }
+                    units seconds;
+                    description
+                         "The value of the timer that defines the period
+                              after which the Authenticator will reauthenticate the Supplicant.";
+                }
+
+                leaf max_users_per_port {
+                    type uint8 {
+                        range 1..48 {
+                            error-message "max users per port value must be in range of 1-48.";
+                            error-app-tag max-users-per-port-invalid;
+                        }
+                    }
+                    description
+                        "Maximum number of clients that can be authenticated
+                             on the port. This is applicable only for multi-auth host mode.";
+                }
+
+                leaf-list method_list {
+                    type auth_order_enumeration;
+                    description
+                        "Enables configuration of authmgr authentication methods order.";
+                }
+
+                leaf-list priority_list {
+                    type auth_priority_enumeration;
+                     description
+                         "Enables configuration of authmgr authentication methods priority.";
+                }
+
+                leaf port_pae_role {
+                    type port_role_enumeration;
+                    description
+                        "Enables configuration of dot1x port's pae role.
+                          Note: Enabling PAC on the port will revert all switchport configurations on the
+                          port,
+                          if port control mode is auto/force-unauthorized and port pae role is
+                          authenticator.";
+                }
+            }
+        }
+
+        container MAB_PORT_CONFIG {
+
+            list MAB_PORT_CONFIG_LIST {
+                key "port";
+
+                leaf port {
+                    type leafref {
+                        path "/prt:sonic-port/prt:PORT/prt:PORT_LIST/prt:name";
+                    }
+                    description
+                        "Name of the interface on which mab gets applied.";
+                }
+
+                leaf mab {
+                    type enumeration {
```

Comment:
https://github.com/sonic-net/sonic-buildimage/blob/master/src/sonic-yang-models/yang-templates/sonic-types.yang.j2#L179
Can you use this type?